### PR TITLE
Migrate interfaces from core repo

### PR DESCRIPTION
**Pull request by ChaoticWalrus**
Sister PR to https://github.com/Layr-Labs/eigenlayer-contracts/pull/294
~~TODO: update the submodule commit that this repo is using. I got this compiling with tests running fine locally though!~~
Edit: looks like this is compiling + passing CI, am marking as ready for review

### CODE REVIEW

File: src/interfaces/IServiceManager.sol [hunk 1]
```diff
@@ -1,4 +1,39 @@
...
+
+    /// @notice proxy call to the slasher, recording a stake update
+    function recordStakeUpdate(address operator, uint32 updateBlock, uint32 serveUntilBlock, uint256 prevElement) external;
+
+    /// @notice proxy call to the slasher, recording a final stake update (on operator deregistration)
+    function recordLastStakeUpdateAndRevokeSlashingAbility(address operator, uint32 serveUntilBlock) external;
+
+    /// @notice Returns the latest block until which operators must serve (could be in the past or future).
+    /// @dev this should be called and the response passed to the recordStakeUpdate functionss' serveUntilBlock parameter
+    function latestServeUntilBlock() external view returns (uint32);
```
Comment (samlaf) [inline src/interfaces/IServiceManager.sol:35]: Can we remove all of these from the interface? Slasher interface is likely going to change, so there's no point in having these. We still want the AVS teams to use this interface for the upcoming M2 testnet release, so need to make an interface that makes sense.

**Review by samlaf — COMMENTED** [review]
Can we remove all stakeUpdate related functionality from the IServiceManager interface? We can keep the `freezeOperator` call since that one is unlikely to change, but all the rest feels like will most likely change.

Will then need to update the ServiceManagerBase to remove all of this recordStakeUpdate proxy calls. We want this to release incredible-squaring and have a consistent story to tell when trying to onboard AVSs onto our testnet.

File: src/interfaces/IServiceManager.sol (hunk 1 above)
Comment (ChaoticWalrus) [reply src/interfaces/IServiceManager.sol:35]: I'm OK with that but would prefer grouping it into a separate PR. This PR is really just aimed at migrating the existing interfaces.

Comment (samlaf) [reply src/interfaces/IServiceManager.sol:35]: Fine with me. Can we get this in tomorrow?

Comment (ChaoticWalrus) [reply src/interfaces/IServiceManager.sol:35]: > Can we get this in tomorrow?

I'm game 👍

**Review by stevennevins — APPROVED** [review]

File: src/interfaces/IServiceManager.sol (hunk 1 above)
Comment (stevennevins) [reply src/interfaces/IServiceManager.sol:35]: https://github.com/Layr-Labs/eigenlayer-middleware/issues/33

**Pull request merged**